plugins/module_utils/data_merge_utils.py

```diff
@@ -0,0 +1,180 @@
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2022, DEMAREST Maxime <maxime@indelog.fr>
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+from copy import deepcopy
+from enum import Enum
```

Comment:
Please note that enums are not supported in Python 2 (which this collection will still support for some time).